pkgs/development/libraries/bashup-events/generic.nix

```diff
@@ -0,0 +1,90 @@
+{
+  # general
+  lib
+, callPackage
+, runCommand
+, resholvePackage
+, fetchFromGitHub
+, bash
+, shellcheck
+, doCheck ? true
+, doInstallCheck ? true
+  # variant-specific
+, variant
+, version
+, branch
+, rev
+, hash
+, fake ? false
+, keep
+}:
+let
+  # extracting this so that it's trivial to test in other shells
+  installCheck = shell:
+    ''
+      echo "testing bashup.events in ${shell}"
+      ${shell} <<'EOF'
+      source $out/bin/bashup.events
+      neat(){
+        echo $0: Hi from event \'test event\'. I can have both $1 and $2 arguments.
+        exit 0
+      }
+      event on "test event" @2 neat curried
+      echo event registered
+      event emit "test event" runtime
+      exit 1 # fail if emitting event didn't exit clean
+      EOF
+    '';
+
+in
+resholvePackage rec {
+  # bashup.events doesn't version yet but it has two variants with
+  # differing features/performance characteristics:
+  # - branch master: a variant for bash 3.2+
+  # - branch bash44: a variant for bash 4.4+
+  pname = "bashup-events${variant}";
+  # should be unstable-YYYY-MM-DD
+  inherit version;
```

Comment:
versions should start with a number always. Typically ISO date + git short rev. Otherwise it's not idempotent with `builtins.parseDrvName`.
```
nix-repl> builtins.parseDrvName "bashup-events-unstable-2020-0
1-01"epl> builtins.parseDrvName "bashup-events-unstable-2020-01- 
{ name = "bashup-events-unstable"; version = "2020-01-01"; }
```

Reply:
moved to pname in https://github.com/NixOS/nixpkgs/pull/107182/commits/168b130e2bd394a915eb3e200d98d2501bffa88b

Reply:
Then why do we have so many packages which gave unstable in the version?